ldclient_test.go

```diff
@@ -30,3 +40,90 @@ func TestOfflineModeAlwaysReturnsDefaultValue(t *testing.T) {
 		t.Errorf("Offline mode should return default value, but doesn't")
 	}
 }
+
+type evaluateTestData struct {
+	on            bool
+	deleted       bool
+	featureKey    string
+	expectedValue interface{}
+	expectError   bool
+}
+
+func TestEvaluate(t *testing.T) {
+	testData := []evaluateTestData{
```

Comment:
Suggestion: Specify the test flags in JSON and have the tests unmarshal the JSON into the `FeatureFlag` struct. I think that will be much easier to update and maintain, and you'll be much better able to re-use that JSON in different SDKs.